spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java

```diff
@@ -604,7 +604,18 @@ public InterpreterResult interpretInput(String[] lines) {
     out.reset();
     Code r = null;
     String incomplete = "";
-    for (String s : linesToRun) {
+
+    for (int l = 0; l < linesToRun.length; l++) {
+      String s = linesToRun[l];      
+      // check if next line starts with "." (but not ".." or "./") it is treated as an invocation
```

Comment:
Thanks for feedback. I think both unclosed parenthetical and end with an infix method already taken care of by detecting incomplete statements (https://github.com/apache/incubator-zeppelin/blob/master/spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java#L622)